### DISCUSSION TITLE
How to turn off progress bar when using perform_experiments()

### DISCUSSION THREAD

**Discussion by EwoutH** [quaquel/EMAworkbench#305] [Q&A]
For benchmarking purposes, I want to turn of the tqdm progress bar (minimize CLI output). I would expect that `reporting_frequency=None` should do the trick. I also tried other values of `log_progress` and `reporting_interval`, but I can't get it to work. 

(CC @quaquel)

**Comment by quaquel** [comment]
There is no way to turn it of at the API level. You would need to comment out the relevant code in [ProgressTrackingMixIn](https://github.com/quaquel/EMAworkbench/blob/81a1f499ae314f50849a9b78fcf4eb0dbd14e250/ema_workbench/em_framework/util.py#L251)

**Reply by EwoutH** [reply]
Would it be useful to add a argument for that? `show_progress_bar`, default `True`?

**Comment by quaquel** [comment]
Looking at the code in more detail, at the level of the mixin, there is a Boolean for this (`log_progress`).  It is also available on `perform_experiment`. The functionality might be there already, and I forgot about it.

**Reply by EwoutH** [reply]
According to the docstring:
https://github.com/quaquel/EMAworkbench/blob/81a1f499ae314f50849a9b78fcf4eb0dbd14e250/ema_workbench/em_framework/callbacks.py#L56-L58

And the code:

https://github.com/quaquel/EMAworkbench/blob/81a1f499ae314f50849a9b78fcf4eb0dbd14e250/ema_workbench/em_framework/util.py#L302-L309

It turns off the progress bar, but enables logs/prints. I think it might be useful to be able to turn both off.

**Reply by quaquel** [reply]
I am not sure why you would want to explicitly control individual log messages. I agree that the current setup already violates this. So, perhaps we should always log at the debug level, while using the boolean to only control the progress bar.
